pkg/csi/service/vanilla/controller.go

```diff
@@ -693,7 +710,7 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 	if containerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIMigration) {
 		// Migration feature switch is enabled
 		if volumePath != "" {
-			req.VolumeId, err = volumeMigrationService.GetVolumeID(ctx, volumePath)
+			req.VolumeId, err = volumeMigrationService.GetVolumeID(ctx, migration.VolumeSpec{VolumePath: volumePath})
```

Comment:
Any reason why we dont set the StoragePolicyID here?

Reply:
In `ControllerUnpublishVolume`, we do not get SPBM policy name as we get it in `ControllerPublishVolume` using `VolumeContext`.

**`ControllerUnpublishVolume` Request**

> 2020-07-28T18:37:49.932064332Z {"level":"info","time":"2020-07-28T18:37:49.92822473Z","caller":"vanilla/controller.go:691","msg":"ControllerUnpublishVolume: called with args {VolumeId:[vsanDatastore] 56631b5f-627a-7a01-5b88-02009d31bab6/kubernetes-dynamic-pvc-b1792c12-35f1-4cfd-946c-7ef84643cdd4.vmdk NodeId:k8s-node2 Secrets:map[] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}","TraceId":"1d56da6d-c0fd-4971-b8de-f2c33f5ffdc5"}


**`ControllerPublishVolume` Request**

> 2020-07-28T18:36:32.832106394Z {"level":"info","time":"2020-07-28T18:36:32.825082689Z","caller":"vanilla/controller.go:577","msg":"ControllerPublishVolume: called with args {VolumeId:[vsanDatastore] 56631b5f-627a-7a01-5b88-02009d31bab6/kubernetes-dynamic-pvc-b1792c12-35f1-4cfd-946c-7ef84643cdd4.vmdk NodeId:k8s-node2 VolumeCapability:mount:<fs_type:\"ext4\" > access_mode:<mode:SINGLE_NODE_WRITER >  Readonly:false Secrets:map[] VolumeContext:map[storagepolicyname:vSAN Default Storage Policy] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}","TraceId":"0f0b7fba-a6d2-49bb-86b6-d88afe67444a"}



Also by this time volume is already registered with CNS, and Pod is up and running. In `ControllerUnpublish` mostly we expect volume migration CR present in the ETCD and the in-memory cache is pre-populated with path to ID mapping.

Reply:
> Also by this time volume is already registered with CNS

This may not be true if the first call back we get for the volume after CSI migration is ControllerUnpublishVolume.

Reply:
The very first call we get is in the syncer for PV update when migration annotation is added.

Also we have a node drain requirement for migration - https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/csi-migration.md#node-drain-requirement to keep volume lifecycle workflow, either with in-tree or with CSI driver. 

We are not supporting the case, where volume is attached using in-tree VCP, but need to be detached with CSI Driver.

Reply:
Can we add some doc in ControllerUnpublishVolume to state that this will never be the first call back for vmdk registration with CNS for so and so reasons? This will help to understand why storage policy is optional in this case.

Reply:
providing the following doc.

```
// ControllerUnpublishVolume will never be the first call back for vmdk registration with CNS.
// Here in the migration.VolumeSpec, we do not supply SPBM Policy name.
// Node drain is the pre-requisite for volume migration, so volume will be registered with SPBM policy
// during ControllerPublish if metadata-syncer fails to register volume using associated SPBM Policy.
// for ControllerUnpublishVolume we anticipate volume is already registered with CNS, and volumeMigrationService
// should return volumeID for requested VolumePath
```